src/shared/menu/plugin.ts

```diff
@@ -112,7 +112,18 @@ export class MenuView implements PluginView {
             const key = (target as HTMLElement).dataset.key;
 
             e.preventDefault();
-            view.focus();
+
```

Comment:
```suggestion
            const isMouseEvent = e.detail > 0; // See https://developer.mozilla.org/en-US/docs/Web/API/UIEvent/detail

            // Move focus to the editor exclusively for mouse clicks
            // For keyboard events, we keep the focus on the menubar
            if (isMouseEvent) {
                // When the click is from a menuitem, hide the popover
                if (target.getAttribute("role") === "menuitem") {
                    const menuButton = (<HTMLElement>e.target).closest(
                        '[data-controller="s-popover"]'
                    );
                    hidePopover(menuButton);
                }

                // leave the menubar and focus on the editor
                view.focus();
            }
```

The `aria-expanded` attribute is not in the `target` element (which is a menuitem in this case) . I think it's a better idea to handle popover management directly with Stacks APIs. `import { hidePopover } from "@stackoverflow/stacks";`

The suggested snippet worked as expected in my local environment.